picard/formats/asf.py

```diff
@@ -186,8 +186,11 @@ def _save(self, filename, metadata):
         log.debug("Saving file %r", filename)
         file = ASF(encode_filename(filename))
 
-        if config.setting['clear_existing_tags']:
-            file.tags.clear()
+        if config.setting["clear_existing_tags"]:
```

Comment:
For apev2 and id3, yes, we have the same:
```python
if config.setting["clear_existing_tags"]:
    if config.setting["remove_extra_padding"]:
        tags.delete()
    else:
        tags.clear()
```
but for asf, mp4 and vorbis, we have
```python
if config.setting["clear_existing_tags"]:
    if config.setting["remove_extra_padding"]:
        file.delete()
    else:
        file.tags.clear()
```

Reading mutagen source code, it seems possible to change the `file.delete()` call into `file.tags.delete()` but I don't recommend it because it would skip a test.
Even in this case, there still would be a difference between `tags` and `file.tags`.
The best I can see would be a function in File like this:
```python
    def remove_tags(self, clear=lambda : None, delete=lambda : None):
        if config.setting["clear_existing_tags"]:
            if config.setting["remove_extra_padding"]:
                delete()
            else:
                clear()
```
and then, call it accordingly `self.remove_tags(tags.clear, tags.delete)` and `self.remove_tags(file.tags.clear, file.delete)`

But I'm not sure it would be really better. I can do it easily, I let you decide ;)